bookish/Services/auxiliary_functions.py

```diff
@@ -0,0 +1,61 @@
+import hashlib
+import datetime
+from flask import request
+from bookish.models.users import Users
+from bookish.models.books import Books
+from bookish.models.checked_out_books import Checked_Out_Books
+from werkzeug.exceptions import HTTPException
+from bookish.error_handler import *
+
+def hashing_algorithm(string):
+    return hashlib.md5(string.encode('utf-8')).hexdigest()
+
+def date_format(string):
+    date_fmt = '%a, %d %b %Y %H:%M:%S %Z'
+    return datetime.datetime.strptime(string, date_fmt)
+
+def check_auth_token(auth_token):
+    """
+    :param token:
+    :return: true -> token valid
+             false -> token expired/invalid
+    """
+
+    all_users = Users.query.all()
+    current_user = next(user for user in all_users if user.user_name == auth_token['user_name'])

```

Comment:
If we have `return_user_by_user_name` and `return_user_by_user_id` maybe we can also have this extracted into a function.